src/POData/ObjectModel/CynicDeserialiser.php

```diff
@@ -46,6 +46,7 @@ public function processPayload(ODataEntry &$payload)
         for ($i = 0; $i < $numLinks; $i++) {
             $this->processLink($payload->links[$i], $sourceSet, $source);
         }
+        assert($this->isEntryProcessed($payload));
```

Comment:
Because OData spec requires that simple and complex requests must be transactional, assert that the entire supplied payload has been successfully processed or cough and die.